magento/magento2
# Adjust mechanism that moves all scripts to the end of the page

### Description (*)
This PR introduces a similar change to #27211, we had this feature implemented in our custom extension and we eventually had to increase both i`pcre.backtrack_limit` and `pcre.recursion_limit` to something around `1000000` in order for the regular expressions to work for large pages.

### Fixed Issues (if relevant)

1. Possibly https://community.magento.com/t5/Magento-2-x-Technical-Issues/Magento-2-3-3-Move-JS-code-to-the-bottom-of-the-page-Error/td-p/348577
2. https://github.com/magento/magento2/issues/26026 - JsFooterPlugin sets Http Response content empty if PCRE limit reached

### Manual testing scenarios (*)

1. Change **Move JS code to the bottom of the page** to **Yes** under Stores->Configuration->Advanced->Developer.
2. Refresh the caches and open the shopfront.
3. Verify that all script tags are located in the same order right before the end of body tag.

### Questions or comments

I am unable to find any tests for this feature and I'm not experienced enough with PHP to write one from scratch so it would be great if someone contributed some tests here. Thanks!
### Contribution checklist (*)
